great_expectations/datasource/fluent/spark_file_path_datasource.py

```diff
@@ -1,87 +1,481 @@
+    )
+    # maxMalformedLogPerPartition: Optional[Union[int, str]] = None,
+    max_malformed_log_per_partition: Optional[Union[int, str]] = Field(
+        None, alias="maxMalformedLogPerPartition"
+    )
+    # mode: Optional[str] = None,
+    mode: Optional[Literal["PERMISSIVE", "DROPMALFORMED", "FAILFAST"]] = None
+    # columnNameOfCorruptRecord: Optional[str] = None,
+    column_name_of_corrupt_record: Optional[str] = Field(
+        None, alias="columnNameOfCorruptRecord"
+    )
+    # multiLine: Optional[Union[bool, str]] = None,
+    multi_line: Optional[Union[bool, str]] = Field(None, alias="multiLine")
+    # charToEscapeQuoteEscaping: Optional[str] = None,
+    char_to_escape_quote_escaping: Optional[str] = Field(
+        None, alias="charToEscapeQuoteEscaping"
+    )
+    # samplingRatio: Optional[Union[float, str]] = None,
+    sampling_ratio: Optional[Union[float, str]] = Field(None, alias="samplingRatio")
+    # enforceSchema: Optional[Union[bool, str]] = None,
+    enforce_schema: Optional[Union[bool, str]] = Field(None, alias="enforceSchema")
+    # emptyValue: Optional[str] = None,
+    empty_value: Optional[str] = Field(None, alias="emptyValue")
+    # locale: Optional[str] = None,
+    locale: Optional[str] = None
+    # lineSep: Optional[str] = None,
+    line_sep: Optional[str] = Field(None, alias="lineSep")
+    # pathGlobFilter: Optional[Union[bool, str]] = None,
+    # Inherited from _SparkGenericFilePathAssetMixin
+    # recursiveFileLookup: Optional[Union[bool, str]] = None,
+    # Inherited from _SparkGenericFilePathAssetMixin
+    # modifiedBefore: Optional[Union[bool, str]] = None,
+    # Inherited from _SparkGenericFilePathAssetMixin
+    # modifiedAfter: Optional[Union[bool, str]] = None,
+    # Inherited from _SparkGenericFilePathAssetMixin
+    # unescapedQuoteHandling: Optional[str] = None,
+    unescaped_quote_handling: Optional[
+        Literal[
+            "STOP_AT_CLOSING_QUOTE",
+            "BACK_TO_DELIMITER",
+            "STOP_AT_DELIMITER",
+            "SKIP_VALUE",
+            "RAISE_ERROR",
+        ]
+    ] = Field(None, alias="unescapedQuoteHandling")
+
+    # vvv Docs <> Source Code mismatch
+    # The following parameters are mentioned in https://spark.apache.org/docs/latest/sql-data-sources-csv.html
+    # however do not appear in the source code https://github.com/apache/spark/blob/v3.4.0/python/pyspark/sql/readwriter.py#L604
+    # prefer_date: bool = Field(True, alias="preferDate")
+    # timestamp_ntz_format: str = Field(
+    #     "yyyy-MM-dd'T'HH:mm:ss[.SSS]", alias="timestampNTZFormat"
+    # )
+    # enable_date_time_parsing_fallback: bool = Field(
+    #     alias="enableDateTimeParsingFallback"
+    # )
+    # ^^^ Docs <> Source Code mismatch
 
     class Config:
         extra = pydantic.Extra.forbid
         allow_population_by_field_name = True
+        # pyspark types (from pyspark_types) are not currently validated
+        arbitrary_types_allowed = True
```

Comment:
Can this be removed?

Reply:
@anthonyburdi If the tests pass, then why not?!